pkg/api/v1beta3/types.go

```diff
@@ -509,6 +509,33 @@ type EnvVar struct {
 	Name string `json:"name" description:"name of the environment variable; must be a C_IDENTIFIER"`
 	// Optional: defaults to "".
 	Value string `json:"value,omitempty" description:"value of the environment variable; defaults to empty string"`
+	// Optional: specify a source for the value of the EnvVar.
+	*EnvVarSource `json:"source,omitempty" description:"source for the environment variable's value; cannot be used if value is set"`
+}
+
+// EnvVarSource represents a source for the value of an EnvVar.
+// Only one of its members may be specified.
+type EnvVarSource struct {
+	// ObjectInfo represents a downward API request for a field of the object.
+	//
+	// Valid fields of this reference are:
+	//
+	// 1. APIVersion
+	// 2. FieldPath
+	//
+	// No other fields may be specified.
+	//
+	// Currently ObjectInfo may be used to address the Name and Namespace
+	// only.
+	ObjectInfo *ObjectReference `json:"objectInfo,omitempty" description:"describes a field of the pod to use the value of; only APIVersion and FieldPath are accepted fields"`
```

Comment:
Yes, like selfLink, we should really populate it based on the version requested, except in the case of cross-prefix references.